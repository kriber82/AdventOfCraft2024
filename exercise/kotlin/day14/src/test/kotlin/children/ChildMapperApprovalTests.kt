package children

import children.ChildMapper
import children.db2.X5T78
import org.approvaltests.Approvals
import org.junit.jupiter.api.Test
import java.util.*

class ChildMapperApprovalTests {
    val mapper = ChildMapper.INSTANCE

    @Test
    fun shouldMapX5T78ToChildForAGirl() {
        val db2Child = X5T78(
            id = UUID.fromString("c253ae9c-a44c-405c-9e00-7eac99b81e60").toString(),
            n1 = "Alice",
            n2 = "Marie",
            n3 = "Smith",
            cityOfBirthPc = "Paradise",
            personBd = "19/03/2017",
            salutation = "Girl",
            stNum = "123",
            stName = "Sunny Street",
            stC = "Paradise",
            stCid = "99"
        )

        val child = mapper.toDto(db2Child)

        Approvals.verify(child)
    }

}