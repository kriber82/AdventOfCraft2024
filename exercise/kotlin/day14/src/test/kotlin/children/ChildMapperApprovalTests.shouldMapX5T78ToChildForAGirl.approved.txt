Child(id=c253ae9c-a44c-405c-9e00-7eac99b81e60, firstName=Alice, middleName=Marie, lastName=Smith, birthCity=Paradise, birthDate=19/03/2017, gender=Girl, address=Address(number=123, street=Sunny Street, city=Paradise, countryId=99))